images/golang-builder/wrapper/main.go

```diff
@@ -161,3 +165,18 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+// getJUnitFileName returns the next available file name for files in the given
+// directory with the naming pattern of junit_<x>.xml where <x> is an increasing
+// integer
+func getJUnitFileName(dir string) string {
+	i := 0
```

Comment:
This is going to be a signed int and as such could overflow and wrap around if no free filenames are available...

I feel we should either use something like a timestamp that can't clash or guard against the case where we've run out of numbers.

Reply:
> This is going to be a signed int and as such could overflow and wrap around if no free filenames are available...

You would have to run `go test` a lot of times within a single CI pipeline for that to happen though, would you be more comfortable if I explicitly made it an `int64`? Alternatively, as you suggest, I can look into timestamping it, but that doesn't guarantee uniqueness strictly